src/Tizen.Wearable.CircularUI.Forms.Renderer/Shell/NavigationDrawer.cs

```diff
@@ -0,0 +1,312 @@
+            new Animation((progress) =>
+            {
+                _outterBox.Opacity = opacity - (int)(progress * opacity);
+
+            }).Commit(this, "FadeOut", length: length, finished: (p, e) =>
+            {
+                _outterBox.Opacity = 0;
+                _outterBox.Hide();
+                tcs.SetResult(true);
+            });
+
+            return await tcs.Task;
+        }
+
+        protected override IntPtr CreateHandle(EvasObject parent)
+        {
+            _gestureBox = new Box(parent);
+            return _gestureBox.Handle;
+        }
+
+        void Initialize()
+        {
+            AlignmentX = -1;
+            AlignmentY = -1;
+            WeightX = 1;
+            WeightY = 1;
+
+            _gestureBox.RepeatEvents = true;
+
+            _outterBox = new Box(this);
+            _outterBox.Show();
+
+            _contentBox = new Box(this);
+            _contentBox.PropagateEvents = false;
+
+            _label = new ElmSharp.Label(_outterBox)
+            {
+                //TODO remove it
+                BackgroundColor = EColor.Pink,
+            };
+
+            _contentBox.Show();
+            _label.Show();
+
+            _outterBox.PackEnd(_contentBox);
+            _outterBox.PackEnd(_label);
+
+            _label.StackAbove(_contentBox);
+
+            _gestureLayer = new GestureLayer(_gestureBox);
+            _gestureLayer.Attach(_gestureBox);
+
+            _gestureLayer.SetMomentumCallback(GestureLayer.GestureState.Start, OnContentDragStarted);
+            _gestureLayer.SetMomentumCallback(GestureLayer.GestureState.End, OnContentDragEnded);
+            _gestureLayer.SetMomentumCallback(GestureLayer.GestureState.Abort, OnContentDragEnded);
+
+            _dragGestureLayer = new GestureLayer(_outterBox);
+            _dragGestureLayer.Attach(_outterBox);
+
+            _dragGestureLayer.SetMomentumCallback(GestureLayer.GestureState.Move, OnDrawerDragged);
+            _dragGestureLayer.SetMomentumCallback(GestureLayer.GestureState.End, OnDrawerDragEnded);
+            _dragGestureLayer.SetMomentumCallback(GestureLayer.GestureState.Abort, OnDrawerDragEnded);
+
+            _gestureBox.SetLayoutCallback(OnLayout);
+            _outterBox.SetLayoutCallback(OnContentLayout);
+
+            RotaryEventManager.Rotated += OnRotateEventReceived;
+        }
+
+        async void OnRotateEventReceived(EventArgs args)
+        {
+            if (_fadeInCancelTokenSource != null)
+            {
+                _fadeInCancelTokenSource.Cancel();
+            }
+
+            _fadeInCancelTokenSource = new CancellationTokenSource();
+            var token = _fadeInCancelTokenSource.Token;
+
+            if (!_isOpen)
+            {
+                await HideAsync(_outterBox);
+
+                _ = ShowAsync(_outterBox, cancelltaionToken:token);
+            }
+        }
+
+        void OnLayout()
+        {
+            var geometry = _parent.Geometry;
```

Comment:
```suggestion
            var geometry = Geometry;
```
Should not use parent geometry.

Reply:
Fixed.